src/attr/_compat.py

```diff
@@ -211,15 +211,22 @@ def force_x_to_be_a_cell():  # pragma: no cover
             )
             set_first_freevar_code = types.CodeType(*args)
 
-        def set_closure_cell(cell, value):
-            # Create a function using the set_first_freevar_code,
-            # whose first closure cell is `cell`. Calling it will
-            # change the value of that cell.
-            setter = types.FunctionType(
-                set_first_freevar_code, {}, "setter", (), (cell,)
-            )
-            # And call it to set the cell.
-            setter(value)
+        if sys.version_info >= (3, 11):
```

Comment:
In fact, it works since Python 3.8:

* https://github.com/python/cpython/commit/df8d2cde63c865446468351f8f648e1c7bd45109
* https://github.com/python/cpython/issues/91299#issuecomment-1093947264